charts/elasticsearch_collector/templates/deployment.yaml

```diff
@@ -11,13 +12,17 @@ spec:
       {{- include "if-elasticsearch-collector.selectorLabels" . | nindent 6 }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
       annotations:
+        {{- with .Values.podAnnotations }}
         {{- toYaml . | nindent 8 }}
-      {{- end }}
+        {{- end }}
+        {{- if $.Values.appArmor.enabled }}
+        container.apparmor.security.beta.kubernetes.io/{{ .Chart.Name }}: runtime/default
+        {{- end }}
       labels:
         {{- include "if-elasticsearch-collector.selectorLabels" . | nindent 8 }}
     spec:
+      automountServiceAccountToken: {{ $.Values.automountServiceAccountToken }}
```

Comment:
We can remove this from values.yaml and set it to false because we don't need additional serviceAccounts for those two agents.